docs/source/nested.rst

```diff
@@ -39,6 +55,15 @@ nested_tensor([
   tensor([3., 4., 5., 6., 7.], device='cuda:0')
 ])
 
+Autograd support is available by setting the `requires_grad` attribute. There is
+a subtlety here where the usual `requires_grad` kwarg for constructors is not supported for
+the `nested_tensor` constructor.
+
+>>> nt.requires_grad_(True)
+
+The tensors passed to the constructor can also require gradients, and autograd will
```

Comment:
I think this is an issue. We should align `torch.nested_tensor` with `torch.tensor` and `torch.as_tensor`. [`torch.tensor`](https://pytorch.org/docs/master/generated/torch.tensor.html#torch.tensor) constructs a leaf Tensor with no autograd history. As such `torch.nested_tensor` should also construct a leaf Tensor. However, the user can use `torch.as_nested_tensor` to get autograd, just like they do with [`torch.as_tensor`](https://pytorch.org/docs/master/generated/torch.as_tensor.html#torch.as_tensor).

Reply:
cc @drisspg @albanD

Reply:
Given the way the `torch.nested_tensor` constructor is defined (as a native_function) would it be possible to add the `requires_grad` kwarg to `torch.tensor`? Or would this be done by wrapping in a python class?

Reply:
It's possible to add kwargs to native functions, so we could add `requires_grad` if we wanted.